framework_tvb/tvb/core/services/flow_service.py

```diff
@@ -280,6 +280,9 @@ def get_launchable_algorithms_for_datatype(self, datatype, categories):
             if not filter_chain or filter_chain.get_python_filter_equivalent(datatype):
                 filtered_adapters.append(stored_adapter)
 
+        if filtered_adapters.__len__() == 0:
+            raise OperationException("There are no valid adapters")
+
```

Comment:
I do not think this exception should be thrown. It is possible that we have no operation compatible with the current DT